src/System.Collections/src/System/Collections/Generic/QueueDebugView.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+
+namespace System.Collections.Generic
+{
+    internal sealed class QueueDebugView<T>
```

Comment:
Sure - I've just done this for HashSet and SortedSet.  I didn't do this for Queue and Stack, since they don't implement the generic ICollection<> interface.  The extension method for ToArray(this IEnumerable<>) is up in System.Linq.